src/main/java/org/passay/LengthComplexityRule.java

```diff
@@ -335,69 +329,23 @@ public boolean includes(final Interval i)
      */
     public boolean intersects(final Interval i)
     {
-      return includes(i.getLowerBoundClosed()) || includes(i.getUpperBoundClosed());
-    }
-
-
-    /**
-     * Returns the closed lower bound for this interval.
-     *
-     * @return  closed lower bound
-     */
-    private int getLowerBoundClosed()
-    {
-      return lowerBound.isClosed() ? lowerBound.value : lowerBound.value + 1;
-    }
-
-
-    /**
-     * Returns the closed upper bound for this interval.
-     *
-     * @return  closed upper bound
-     */
-    private int getUpperBoundClosed()
-    {
-      return upperBound.isClosed() ? upperBound.value : upperBound.value - 1;
-    }
-
-
-    @Override
-    public boolean equals(final Object o)
-    {
-      if (o == this) {
-        return true;
-      }
-      if (o != null && getClass() == o.getClass())  {
-        final Interval other = (Interval) o;
-        return lowerBound.equals(other.lowerBound) && upperBound.equals(other.upperBound);
-      }
-      return false;
-    }
-
-
-    @Override
-    public int hashCode()
-    {
-      return Objects.hash(lowerBound, upperBound);
+      return includes(i.lower) || includes(i.upper);
```

Comment:
I think the previous impl makes more sense. Or `boundsPattern.hashCode()` in the new impl.

Reply:
If you're referring to hashCode, the same answer applies as to equals above. I can add it, but it's not needed. If you meant something else rather than hashCode, please elaborate...